File: Sorting Algorithms/Python Implementation/Selection_Sort.py
class selectionsort:
    def __init__(self,x):
        self.x=x
    def selectionsort(self):
        x=self.x
        for i in range(len(x)):
            a=0
            b=(len(x)-1)-i
            for j in range(1,b):
                if(x[a]<=x[j]):
                    a=j
            temp=x[a]
            x[a]=x[b]
            x[b]=temp
        return x
x=[9,3,6,1,7,3,2]
result = selectionsort(x)
print(result.selectionsort())

File: Sorting Algorithms/Java Implementation/Selection_Sort.java
import java.util.*;
public class test{
	public static int[] selectionsort(int[] x) {
		for(int i=x.length-1;i>0;i--) {
			int a=0;
			for(int j=1;j<=i;j++) {
				if(x[a]<=x[j]) {
					a=j;
				}
			}
			int temp = x[a];
			x[a]=x[i];
			x[i]=temp;
		}
		return x;
	}
	public static void main(String args[]) {
		int[] x = {9,3,6,1,11,3,2,10};
		int [] result = selectionsort(x);
		System.out.println("The sorted array using selection sort is "+Arrays.toString(result));		
	}
}
